### DISCUSSION TITLE
Questions about interpolation and basis

### DISCUSSION THREAD

**Discussion by lucasgrjn** [kinnala/scikit-fem#1085] [Q&A] [answered]
Hi first, of all thanks for scikit-fem! It is really a great package :)
I try to dig and learn into the FEM, but I have few questions regarding some uses.

1. Interpolation and dimensions (from docs)
```python
import numpy as np
from skfem import *

m = MeshQuad().refined(2)
basis = FacetBasis(m, ElementQuad1())
u0 = lambda x: x[0] ** 3 * x[1] ** 3
u0t = basis.project(u0)
u0i = basis.interpolate(u0t)
```
Why in this code `u0i` will get a shape `(#elems, 3)`.
I get that `u0t` will have a shape `(25,)` because it will be projected on the nodes of the mesh.
But I don't understand with `u0i` has a 3-shape (here `(16,3)`. Since the basis is defined by ElementQuad1, should it not be a `(16,4)`?

2. Regarding the use of ElementDG
I saw in the code the use as `ElementDG` like in example 46 as:
```python
plotbasis = Ebasis.with_element(ElementDG(ElementTriP2()))
```
What is the purpose of `ElementDG` and why is it used like that here?

Thank in advance for your help,
And thanks again for the amazing work on scikit-fem!
Lucas

**Comment by kinnala** [comment]
1. Here 3 is the number of integration points per facet, in this case on the boundary of the mesh since you are using FacetBasis. By default, FacetBasis calculates the number of points required automatically which is not always ideal. You can change the integration order using FacetBasis(..., intorder=5). You can also specify your own integration rule but this is advanced usage: FacetBasis(..., quadrature=(points, weigths)). Quadrature points should be for the reference domain [0, 1].

**Reply by kinnala** [reply, marked as answer]
2. ElementDG will turn any element into another element. All (reference element) shape functions are reused but any continuity from one element to another is removed.

**Reply by kinnala** [reply]
This is not my example, so not entirely sure why it is used here, but perhaps the quantity which is being visualized is discontinuous over the element boundaries?

**Reply by lucasgrjn** [reply]
Not sure for ex46, but it would make sense!

Thanks for the responses :)
